src/js/components/Pagination/Pagination.js

```diff
@@ -176,9 +178,41 @@ const Pagination = forwardRef(
       ...navProps[control],
     }));
 
+    const itemsBegin = step * (activePage - 1) + 1;
+    const itemsEnd = itemsBegin + step - 1;
+    let showSummary;
+    if (showSummaryProp === true)
+      showSummary = (
+        <Text>
+          Showing {itemsBegin} - {itemsEnd} of {numberItems}
+        </Text>
+      );
+    else showSummary = showSummaryProp;
+
+    // only apply these if the caller hasn't already specified their own theming
+    // otherwise, let the caller control
+    const showSummaryStyles =
+      showSummary && !theme.pagination.container
+        ? {
+            align: 'center',
+            fill: 'horizontal',
+            direction: 'row',
+            justify: 'between',
+          }
+        : undefined;
+
     return (
-      <StyledPaginationContainer {...theme.pagination.container} {...rest}>
-        <Nav a11yTitle={a11yTitle || 'Pagination Navigation'} ref={ref}>
+      <StyledPaginationContainer
+        {...theme.pagination.container}
+        {...showSummaryStyles}
+        {...rest}
+      >
+        {showSummary}
+        <Nav
+          a11yTitle={a11yTitle || 'Pagination Navigation'}
+          ref={ref}
+          {...theme.pagination.nav}
```

Comment:
What is the intent behind this? What needs to be handled here which could not be addressed by styling the Container or the Controls?

Reply:
For the Custom Summary example I showed with the Select component that allows the user to change # results/per page, I needed the structure to be:
```
<Box direction="row" justify="between">
   <Box flex (contains summary) />   <Box with Select />  <Box flex (contains pagination controls) />
</Box>
```
The flex props allow the side boxes to grow/shrink while leaving the middle box centered. However, I could not put flex: true on `theme.pagination.controls` because this is the box contained inside the `nav`, we need the nav itself to be able to grow.

I also can't move the `theme.pagination.controls` styling up to the nav because then the direction, alignment, spacing, etc will not be applied to the pagination buttons since they are not a direct child of nav.

Open to exploring other approaches but this was the reason I felt I needed to add it.

Reply:
Thanks for the explanation. I'll explore this scenario a bit more.